Initial bot settings

mdrgoncalves opened the pull request:
Changes throughout the repository.

- Set initial bot settings for its full operation.
- Created commands: Hello.

mdrgoncalves committed:
feat: initializing bot

Configuring initial bot settings

mdrgoncalves committed:
feat: adds slash commands configuration

Created a configuration that allows and facilitates the scalability of commands via slash

mdrgoncalves committed:
feat: added initial commands

mdrgoncalves committed:
refactor: changes for a more accurate code

mdrgoncalves committed:
refactor: adds interfaces folder

mdrgoncalves committed:
feat: adds dev command with nodemon

lcsmuller commented on src/commands/FxTweet.ts:
```diff
@@ -0,0 +1,44 @@
+import {
+  CommandInteraction,
+  Client,
+  ApplicationCommandType,
+  ApplicationCommandOptionType,
+  InteractionResponse,
+} from 'discord.js';
+
+import { Command } from '../interfaces/Command';
+
+export const FxTweet: Command = {
+  name: 'fxtweet',
+  description: 'Fix a tweet video embed',
+  type: ApplicationCommandType.ChatInput,
+  options: [
+    {
+      name: 'tweet',
+      description: 'Twitter URL',
+      type: ApplicationCommandOptionType.String,
+      required: true,
+    },
+  ],
+  run: (
+    client: Client,
+    interaction: CommandInteraction,
+  ): Promise<InteractionResponse<boolean>> => {
```
Nit: `Command` already infers callback's types!
```suggestion
import { ApplicationCommandType, ApplicationCommandOptionType } from 'discord.js';

import { Command } from '../interfaces/Command';

export const FxTweet: Command = {
  name: 'fxtweet',
  description: 'Fix a tweet video embed',
  type: ApplicationCommandType.ChatInput,
  options: [
    {
      name: 'tweet',
      description: 'Twitter URL',
      type: ApplicationCommandOptionType.String,
      required: true,
    },
  ],
  run: (_, interaction) => {
```

lcsmuller commented on src/interfaces/Command.ts:
```diff
@@ -0,0 +1,5 @@
+import { CommandInteraction, ChatInputApplicationCommandData, Client } from "discord.js";
+
+export interface Command extends ChatInputApplicationCommandData {
+    run: (client: Client, interaction: CommandInteraction) => void;
```
Should it return a `Promise<InteractionResponse<boolean>>` instead?

lcsmuller commented on src/commands/Hello.ts:
```diff
@@ -0,0 +1,26 @@
+import {
+  CommandInteraction,
+  Client,
+  ApplicationCommandType,
+  InteractionResponse,
+} from 'discord.js';
+
+import { Command } from '../interfaces/Command';
+
+export const Hello: Command = {
+  name: 'hello',
+  description: 'Returns a greeting',
+  type: ApplicationCommandType.ChatInput,
+  run: (
+    client: Client,
+    interaction: CommandInteraction,
+  ): Promise<InteractionResponse<boolean>> => {
+    const username = interaction.user?.username;
+    const content = `Ara ara... Hi, ${username}-kun!`;
+
+    return interaction.reply({
+      ephemeral: true,
+      content,
+    });
+  },
+};
```
ditto, also we could turn this into `as const satisfies` and achieve better typing..
```suggestion
import { ApplicationCommandType } from 'discord.js';

import { Command } from '../interfaces/Command';

export const Hello = {
  name: 'hello',
  description: 'Returns a greeting',
  type: ApplicationCommandType.ChatInput,
  run: (_, interaction) => {
    const username = interaction.user?.username;
    const content = `Ara ara... Hi, ${username}-kun!`;

    return interaction.reply({
      ephemeral: true,
      content,
    });
  },
} as const satisfies Command;
```

lcsmuller commented on src/commands/Hello.ts:
```diff
@@ -0,0 +1,26 @@
+    client: Client,
+    interaction: CommandInteraction,
+  ): Promise<InteractionResponse<boolean>> => {
+    const username = interaction.user?.username;
```
can user ever be undefined here? we may want to throw an error in that case

lcsmuller commented on src/listeners/interactionCreate.ts:
```diff
@@ -0,0 +1,25 @@
+import { Client, CommandInteraction, Interaction } from 'discord.js';
+import { Commands } from '../Commands';
+
+export default (client: Client): void => {
+  client.on('interactionCreate', async (interaction: Interaction) => {
+    if (interaction.isCommand()) {
+      await handleSlashCommand(client, interaction);
+    }
+  });
+};
```
Nit: type is inferred automatically, `async` may be omitted
```suggestion
export default (client: Client): void => {
  client.on('interactionCreate', interaction => {
    if (interaction.isCommand()) {
      return handleSlashCommand(client, interaction);
    }
  });
};
```

lcsmuller commented on src/listeners/interactionCreate.ts:
```diff
@@ -0,0 +1,25 @@
+  });
+};
+
+const handleSlashCommand = async (
+  client: Client,
+  interaction: CommandInteraction,
+): Promise<void> => {
+  const slashCommand = Commands.find(
+    command => command.name === interaction.commandName,
+  );
+  if (!slashCommand) {
+    interaction.followUp({ content: 'An error has occurred' });
+    return;
+  }
+
+  slashCommand.run(client, interaction);
+};
```
Unused `async`, but `interaction.followUp` seems to return a Promise:
```suggestion
const handleSlashCommand = (
  client: Client,
  interaction: CommandInteraction,
): Promise<void> => {
  const slashCommand = Commands.find(
    command => command.name === interaction.commandName,
  );
  if (!slashCommand) {
    return interaction.followUp({ content: 'An error has occurred' });
  }

  slashCommand.run(client, interaction);
};
```

lcsmuller commented on src/commands/FxTweet.ts:
```diff
@@ -0,0 +1,44 @@
+
+import { Command } from '../interfaces/Command';
+
+export const FxTweet: Command = {
```
Up to you, but using `satisfies as Command` should provide more accurate typing when hovering the mouse over it:
```ts
const FxTweet: Command
// vs
const FxTweet: {
    name: string;
    description: string;
    type: ApplicationCommandType.ChatInput;
    options: {
        name: string;
        description: string;
        type: ApplicationCommandOptionType.String;
        required: true;
    }[];
    run: (_: Client<...>, interaction: CommandInteraction<...>) => Promise<...>;
}
```

lcsmuller commented on src/commands/FxTweet.ts:
```diff
@@ -0,0 +1,44 @@
+import {
```
Nit: These are all types, so may be imported as such:
```suggestion
import type {
```

lcsmuller commented on src/commands/Hello.ts:
```diff
@@ -0,0 +1,26 @@
+import {
```
ditto

lcsmuller commented on src/commands/Hello.ts:
```diff
@@ -0,0 +1,26 @@
+  InteractionResponse,
+} from 'discord.js';
+
+import { Command } from '../interfaces/Command';
```
ditto

lcsmuller commented on src/index.ts:
```diff
@@ -1,20 +1,17 @@
-import { Client } from 'discord.js';
+import { Client } from "discord.js";
```
ditto

lcsmuller commented on src/listeners/interactionCreate.ts:
```diff
@@ -0,0 +1,25 @@
+import { Client, CommandInteraction, Interaction } from 'discord.js';
+import { Commands } from '../Commands';
```
ditto

lcsmuller commented on src/listeners/ready.ts:
```diff
@@ -0,0 +1,18 @@
+import { Client, ActivityType } from 'discord.js';
+import { Commands } from '../Commands';
```
ditto

mdrgoncalves committed:
chore: update yarn version

mdrgoncalves committed:
refactor: remove FxTweet command and update Hello command

- Removed the FxTweet command from the Commands array in src/Commands.ts
- Deleted the FxTweet command file in src/commands/FxTweet.ts
- Updated the Hello command in src/commands/Hello.ts to remove unused imports and add type annotations
- Updated the Command interface in src/interfaces/Command.ts to include a return type

mdrgoncalves committed:
refactor: remove unused import statements
